sdk/cmake/host-tools.cmake

```diff
@@ -29,6 +29,48 @@ function(setup_host_tools)
         set(HOST_MODULE_SUFFIX ".so")
     endif()
 
+    # Normalize to the same format as our own ARCH, and add one for the VC shell
+    string(TOLOWER "${CMAKE_HOST_SYSTEM_PROCESSOR}" lowercase_CMAKE_HOST_SYSTEM_PROCESSOR)
+    if(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^i[3456]86$")
+        set(HOST_ARCH i386)
+        set(VCVARSALL_ARCH x86)
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
+        set(HOST_ARCH amd64)
+        set(VCVARSALL_ARCH amd64_x86) # x64 host-tools are not happy compiling for x86...
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL arm)
+        set(HOST_ARCH arm)
+        set(VCVARSALL_ARCH arm)
+    else()
+        message(FATAL_ERROR "Unknown host architecture: ${lowercase_CMAKE_HOST_SYSTEM_PROCESSOR}")
+    endif()
+
+    if(ARCH STREQUAL HOST_ARCH)
+        set(HOST_TOOLS_CMAKE_COMMAND "${CMAKE_COMMAND}")
+        message("Not cross-compiling, no special host-tools cmake command")
```

Comment:
Swap these 2 lines.

Reply:
no, because that would mean that the wrong text is added in the wrong case.

Reply:
?? Having `message()` first should not change anything with regard to text/case...

Reply:
I thought you meant swap the comments from the 2 cases where you commented on.